test_suite/cloud/test_aws.py

```diff
@@ -29,6 +29,22 @@ def instance_data_aws_cli(host, instance_data_aws_web):
 
 @pytest.mark.order(2)
 class TestsAWS:
+    @pytest.mark.pub
+    @pytest.mark.run_on(['rhel', 'fedora'])
+    def test_etc_machine_id_permissions(self, host, instance_data):
+        """
+        Check that /etc/machine-id permissions are 444. RHBZ #2221269
+        """
+        machine_id = "/etc/machine-id"
+
+        assert os.access(machine_id, os.R_OK), f"Expected read access for {machine_id}"
```

Comment:
Maybe this can be simplified to just check that the octal permissions are `444`, using testinfra's host module.
I think that os.* will not be run in the remote host but on the runner itself (e.g. locally in Jenkins agent or GitLab runner). For that we need to use `host.run('whatever shell cmd')` or similar.

See this test case as an example:
* https://github.com/osbuild/cloud-image-val/blob/fcc748864776b1363cc165bdfc7b520c360cb35e/test_suite/cloud/test_aws.py#L378C75-L378C75

You can find more information and usages of testinfra modules here:
https://testinfra.readthedocs.io/en/latest/modules.html#testinfra.modules.file.File.mode

Reply:
About execution access and similar, we may have to run shell commands with `host.run()` and capture `stdout` from it.